src/interactive-window/editor-integration/codewatcher.node.ts

```diff
@@ -1021,24 +1021,25 @@ export class CodeWatcher implements ICodeWatcher {
         const nextRunCellLens = this.getNextCellLens(range.start);
 
         if (currentRunCellLens) {
-            // Move the next cell if allowed.
-            if (advance) {
-                const editor = this.documentManager.activeTextEditor;
-                const { newCellOnRunLast } = this.configService.getSettings(editor?.document.uri);
-                if (nextRunCellLens) {
-                    this.advanceToRange(nextRunCellLens.range);
-                } else if (newCellOnRunLast && editor) {
-                    // insert new cell at bottom after current
-                    this.insertCell(editor, currentRunCellLens.range.end.line + 1);
-                }
-            }
-
-            // Run the cell after moving the selection
             if (this.document) {
                 const iw = await this.getActiveInteractiveWindow();
-
-                // Use that to get our code.
                 const code = this.document.getText(currentRunCellLens.range);
+
+                // Move the next cell if allowed.
+                if (advance) {
+                    const editor = this.documentManager.activeTextEditor;
```

Comment:
We don't need the editor object in this case also since this is after the line `await this.getActiveInter...`
its possibel focus moves after that.

Instead we can just change `this.configService.getSettings(this.document.uri)`